packages/contract-call-stream/__tests__/node_env.js

```diff
@@ -68,7 +68,6 @@ describe("contract-call-stream tests in node environment", () => {
         tap(vals => expect(vals).toEqual(["0", "5"])),
         finalize(() => {
           expect.assertions(1);
-          cleanup();
```

Comment:
Shouldn't we cleanup using newBlock$.unsubscribe (I assume that's what it is now)?

Reply:
Yeah, that's correct. Let me give that a shot.

Reply:
@honestbonsai Actually we don't have to, because the `finalize` function happens when:

> Call a function when observable completes or errors

[Source](https://www.learnrxjs.io/operators/utility/finalize.html)

That means the Subscription stops and there's no memory leak. In other words, if `finalize` is running, then the stream has already stopped.

Reply:
Modern doc reference: https://rxjs-dev.firebaseapp.com/api/operators/finalize

Reply:
@honestbonsai notice that `take(2)` is in the pipe, which means that the stream will automatically complete after 2 events have passed through it.

Reply:
@adrianmcli Sounds good to me then